api/v1beta1/disruption_types_test.go

```diff
@@ -137,3 +141,312 @@ var _ = Describe("Check if a target exist into DisruptionStatus targets list", f
 		})
 	})
 })
+
+var _ = Describe("Disruption", func() {
+
+	var (
+		defaultCreationTimestamp = time.Now()
+		notBeforeTime            = defaultCreationTimestamp.Add(time.Minute)
+	)
+
+	DescribeTable("TimeToInject", func(disruptionBuilder *builderstest.DisruptionBuilder, expectedTime time.Time) {
+		// Arrange
+		disruption := disruptionBuilder.WithCreationTime(defaultCreationTimestamp).Build()
+
+		// Action && Assert
+		Expect(disruption.TimeToInject()).To(Equal(expectedTime))
+	},
+		Entry(
+			"should return creationTimestamp if triggers is nil",
+			builderstest.NewDisruptionBuilder(), defaultCreationTimestamp),
+		Entry(
+			"should return triggers.createPods if triggers.inject is nil",
+			builderstest.NewDisruptionBuilder().WithDisruptionTriggers(DisruptionTriggers{
+				CreatePods: DisruptionTrigger{
+					NotBefore: metav1.NewTime(notBeforeTime),
+					Offset:    "",
+				},
+			}), notBeforeTime),
+		Entry(
+			"should return inject.notBefore if set",
+			builderstest.NewDisruptionBuilder().WithDisruptionTriggers(DisruptionTriggers{
+				Inject: DisruptionTrigger{
+					NotBefore: metav1.NewTime(notBeforeTime),
+					Offset:    "1",
+				},
+				CreatePods: DisruptionTrigger{
+					NotBefore: metav1.Time{},
+					Offset:    "2m",
```

Comment:
This case where an explicit inject.notBefore comes _before_ the createPods.offset is unusual, but seems technically allowed by the spec.

Reply:
So maybe we should handle this case? WDYT?

Reply:
It should be done during the validation of the spec?

Reply:
I don't think you need to handle it as part of this PR. If we do change it, yeah I think preventing it in spec validation would be the better call, it indicates a user who is probably making a mistake